platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -50,6 +50,33 @@ Ant has been updated to https://archive.apache.org/dist/ant/RELEASE-NOTES-1.10.1
 
 === Deprecations
 
+[[minimum_daemon_jvm_version]]
+==== Deprecated running the Gradle daemon with JVM versions 16 and earlier
+
+Starting in Gradle 9.0, the Gradle daemon will require JVM 17 or later to run and Gradle API classes will be compiled to target JVM 17 bytecode.
+
+Using Gradle to compile Java code to target JVM version 6 or later is still fully supported.
+The <<building_java_projects.adoc#sec:java_cross_compilation,target JVM version>> of the compiled code can be configured separately from the JVM version used to run the Gradle daemon.
+
+The Gradle Wrapper, Gradle client, Tooling API client, and TestKit client will remain compatible with Java 8 and will be compiled to target JVM 8 bytecode.
+These daemon clients can be configured to launch Gradle daemons with a different JVM version than the one used to run the client:
+
+- Using <<gradle_daemon#sec:daemon_jvm_criteria,Daemon JVM criteria>> (an incubating feature)
+- Setting the `org.gradle.java.home` <<build_environment.adoc#sec:gradle_configuration_properties,Gradle property>>
+- Using the link:{javadocPath}/org/gradle/tooling/ConfigurableLauncher.html#setJavaHome(java.io.File)[ConfigurableLauncher#setJavaHome] method on the Tooling API
+
+Alternatively, the `JAVA_HOME` environment variable can be set to a JVM 17 or newer.
+
+Running Gradle builds with <<gradle_daemon#sec:disabling_the_daemon,--no-daemon>> or using link:{javadocPath}/org/gradle/testfixtures/ProjectBuilder.html[ProjectBuilder] will require a JVM version 17 or later.
+
+We have decided to upgrade the minimum version of Java that Gradle will run on for a number of reasons:
+
+- Gradle's own dependencies are beginning to drop support for older JVM versions. We are concerned about the security risks of not being able to upgrade dependencies due to Java version compatibility constraints.
+- Some of the largest ecosystems that leverage Gradle already require JVM 17 or later.
+- The language improvements between Java 8 and Java 17 are significant. Upgrading improves development and maintenance efficiency within the Gradle codebase.
+
+In upcoming major Gradle releases, we will continue to evaluate the minimum JVM version required to run Gradle.
```

Comment:
I'm not sure what this is trying to say?  Something like we'll re-evaluate if Gradle 10 will require Java 21?

```suggestion
```

Reply:
Yeah like “don’t think this is the last time we’re upgrading the minimum version”

This is in place of defining a more formal compatibility contract